test/sanity/validate-modules/validate-modules

```diff
@@ -782,9 +892,12 @@ def main():
                 with ModuleValidator(path, analyze_arg_spec=args.arg_spec,
                                      base_branch=args.base_branch) as mv:
                     mv.validate()
-                    exit.append(mv.report(args.warnings))
+                    reports.update(mv.report())
 
-    sys.exit(sum(exit))
+    if args.format == 'plain':
+        sys.exit(Reporter.plain(reports, warnings=args.warnings, output=args.output))
+    else:
+        sys.exit(Reporter.json(reports, warnings=args.warnings, output=args.output))
```

Comment:
I see you point. At the very least, it would be useful to have predictable error codes. Perhaps something like this:

0 - no errors found
1 - errors found, JSON successfully written
2 - unexpected failure, JSON not successfully written

I don't see a need for having the error code be dependent on the number of errors found.